src/infra/dex/okx/mod.rs

```diff
@@ -0,0 +1,255 @@
+use {
+    crate::{
+        domain::{dex, eth, order},
+        util,
+    },
+    base64::prelude::*,
+    chrono::SecondsFormat,
+    ethrpc::block_stream::CurrentBlockWatcher,
+    hmac::{Hmac, Mac},
+    hyper::{header::HeaderValue, StatusCode},
+    sha2::Sha256,
+    std::sync::atomic::{self, AtomicU64},
+    tracing::Instrument,
+};
+
+mod dto;
+
+/// Bindings to the OKX swap API.
+pub struct Okx {
+    client: super::Client,
+    endpoint: reqwest::Url,
+    api_secret_key: String,
+    defaults: dto::SwapRequest,
+}
+
+pub struct Config {
+    /// The base URL for the 0KX swap API.
+    pub endpoint: reqwest::Url,
+
+    pub chain_id: eth::ChainId,
+
+    /// OKX project ID to use. Instruction on how to create project:
```

Comment:
Would be nice to group these OKX credentials parameters into a struct to make them a logical unit.